### ISSUE TITLE
Formatting doesn't work in VScode

### ISSUE THREAD

**Issue by davidlindercodes** [prettier/plugin-php#1730]
My setup looks like this: 
VS Code (latest version - no updates available)
Prettier code formatter (6.4.0) - Installed via VS Code extensions 
Prettier/plugin-php  - installed globally using:
`    npm install --global prettier @prettier/plugin-php`

My VS code setting.json file looks like this: 
```
    {
    "editor.codeActionsOnSave": null,
    "editor.defaultFormatter": "esbenp.prettier-vscode",
    "editor.formatOnSave": true
    }
```

Yes when I try to format a .php file, using either format on save or smd-shift-p nothing happens, it has zero functionality. What am I doing wrong here? Is there an installation set that I missed? 
   
 Prettier is currently working as expected for .js files, it is only plug-php that I am having issues with.

**Comment by czosel** [comment]
Did you already try installing the plugin _locally_ in your project? (`npm install prettier @prettier/plugin-php`) It seems that making the prettier-vscode plugin work with global installations of prettier and plugins requires some configuration: https://github.com/prettier/prettier-vscode#prettier-resolution

Also see https://github.com/prettier/plugin-php/issues/1721#issuecomment-840684021 for step-by-step instructions.

**Comment by davidlindercodes** [comment]
Hi Czosel, 

Yes, I tried installing the plugin locally, but the same issue still exists.

**Comment by D1360-64RC14** [comment]
Same problem.

On terminal Prettier formats without problems, but VSCode's extension doesn't recognize the plugin. Even specifying the global or local *node_modules* in `pluginSearchDirs`. Prettier is locally and globally installed.

![image](https://user-images.githubusercontent.com/33406930/150858507-0a854da9-0a76-4876-98f8-fc9fc5194ce1.png)

My `.prettierrc`
```json
{
  "phpVersion": "7.0",
  "tabWidth": 4,
  "useTabs": false,
  "singleQuote": true,
  "trailingCommaPHP": true,
  "pluginSearchDirs": [
    "/HDExterno/Langs/PHP/plugin/html/wp-content/themes/api/node_modules",
    "./node_modules",
    "node_modules"
  ],
  "plugins": [
    "@prettier/plugin-php"
  ],
  "parser": "php"
}
```

My `settings.json`:
```jsonc
    // ...
    "editor.defaultFormatter": "esbenp.prettier-vscode",
    "[php]": {
        "editor.formatOnSave": true
    },
    // ...
```

**Comment by tm1000** [comment]
@D1360-64RC14 setting pluginSearchDirs, plugins and parser is not necessary

**Comment by kjoedion** [comment]
I was also having trouble with the global installation.

Adding this to my vscode user `settings.json` made it work!:

```
"prettier.resolveGlobalModules": true
```

**Comment by tm1000** [comment]
@kjoedion makes sense. For me I install prettier php for every project individually

**Comment by WazzaJB** [comment]
The solutions I found across the various tickets here didn't work for me unfortunately, but appreciate everyones efforts to come up with a solution.

I had to extend the prettier document selectors within VS Code to get it to run on PHP files:

.vscode/settings.json
```
"prettier.documentSelectors": [
    "**/*.{js,jsx,ts,tsx,vue,html,css,scss,less,json,md,mdx,graphql,yaml,yml,php}"
 ]
  ```
  
  I also added `parser` to the prettier config and ended up with this:
 .prettierrc
 ```
 {
  "parser": "php",
  "plugins": ["@prettier/plugin-php"],
  "phpVersion": "8.2"
}
```

All works fine in VS Code for me now. Hopefully this helps someone else.

Related tickets are: https://github.com/prettier/plugin-php/issues/1721, https://github.com/prettier/plugin-php/issues/1640, https://github.com/prettier/prettier-vscode/issues/2564
